src/chocolatey/infrastructure.app/runners/ConsoleApplication.cs

```diff
@@ -85,9 +88,63 @@ public void run(string[] args, ChocolateyConfiguration config, Container contain
                                 }
                             }
                         },
-                        () => command.handle_validation(config),
+                        () =>
+                        {
+                            command.handle_validation(config);
+
+                            var validationResults = new List<ValidationResult>();
+                            var validations = container.GetAllInstances<IValidation>();
+
+                            foreach (var validation in validations)
+                            {
+                                validationResults.AddRange(validation.validate(config));
+                            }
+
+                            var validationErrors = report_validation_summary(validationResults);
+
+                            if (validationErrors != 0)
+                            {
+                                // NOTE: This is intentionally left blank
+                                throw new ApplicationException("");
+                            }
+                        },
                         () => command.help_message(config));
                 });
         }
+
+        private int report_validation_summary(IList<ValidationResult> validationResults)
+        {
+            var successes = validationResults.Count(v => v.Status == ValidationStatus.Success);
+            var warnings = validationResults.Count(v => v.Status == ValidationStatus.Warning);
+            var errors = validationResults.Count(v => v.Status == ValidationStatus.Error);
+
+            this.Log().Info("{0} validations performed. {1} success(es), {2} warning(s), and {3} error(s).".format_with(
+                validationResults.Count,
+                successes,
+                warnings,
+                errors));
+
+            if (warnings != 0)
+            {
+                this.Log().Info("");
+                this.Log().Warn("Warnings:");
+                foreach (var warning in validationResults.Where(p => p.Status == ValidationStatus.Warning).or_empty_list_if_null())
+                {
+                    this.Log().Warn(" - {0}".format_with(warning.Message));
+                }
+            }
+
+            if (errors != 0)
+            {
+                this.Log().Info("");
+                this.Log().Error("Errors:");
+                foreach (var error in validationResults.Where(p => p.Status == ValidationStatus.Error).or_empty_list_if_null())
+                {
+                    this.Log().Error(" - {0}".format_with(error.Message, error.ExitCode));
+                }
+            }
+
+            return errors;

```

Comment:
Overall good summary. However I think I might bump this up so the messaging looks like this:

~~~
5 validations performed. 2 success(es), 1 warning(s), and 2 error(s).
 Warnings:
   - Message here
 Errors:
   - Error message here (exit 350).
   - Next Error message here (exit 2).
~~~

Reply:
If the words "Warnings" and "Errors" do have a line break, they need to be prefixed with "Validation" as in "Validation Warnings"/"Validation Errors"